docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,199 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
+
+:term:`ConfigMap` is chosen as a unified data access and storage media for user
+editable configurations in a MetalK8s cluster based on the above requirements
+for the following reasons:
+
+* Ability to support Update operations on ConfigMap's with CLI and UI easily
+  using our already existing python kubernetes module.
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly.
+* ConfigMaps stored in the :term:`etcd` database which is generally being
+  backed up. This ensures that user settings cannot be lost easily.
+
+
+How it works
+~~~~~~~~~~~~
+
+During Bootstrap, pillar values for the various services will be pre-filled
+with default values. Service pods and deployments will be configured to either
+consume configuration data directly from their ConfigMap or fallback to
+the default pillar values if the service specific ConfigMap was not found.
+
+**Using Salt states**
+
+Once a ConfigMap is updated by the user (say a user changes the number of
+replicas for Prometheus deployments to a new value), then perform the
+following actions:
+
+  - Apply a Salt states that reads the ConfigMap object, validates the schema
+    and checks the new values passed.
+
+  - Finally, we make sure that the updated values are picked up by their
+    respective consumers (this might require Pod restarts for changes to take
+    effect).
+
+Note that, Salt states are used to sync data and update consumers
+of new configurations changes mainly because of the minimum effort it takes to
+setup this flow but the K8s Operator pattern could be use to replace
+configuration synchronization between user defined configurations and
+consumers.
+
+Non-goals
+~~~~~~~~~
+
+- Persisting newly added Grafana dashboards or new Grafana datasources
+  especially for modifications added via the Grafana UI cannot be stored in
+  ConfigMaps.
+  To handle this particular user case, we need to configure Grafana deployments
+  to use persistent volume claims and then document how to create volumes that
```

Comment:
Here you're prescribing *how* a certain feature *not* covered by this design ought to be implemented. That's not OK. Indeed, this document should describe what it *won't* cover, but whether this other feature is implemented using PVs or some other mechanism is to be determined later/elsewhere.

Reply:
True, I will move it to non-goals and clearly state that the current design does not cater for this. No need to mention an implementation choice.